include/onnxruntime/core/framework/allocator.h

```diff
@@ -37,6 +37,8 @@ namespace onnxruntime {
 constexpr const char* CPU = "Cpu";
 constexpr const char* CUDA = "Cuda";
 constexpr const char* CUDA_PINNED = "CudaPinned";
+constexpr const char* CANN = "Cann";
+constexpr const char* CANN_PINNED = "CannPinned";
```

Comment:
just confirming that there really need for pinned memory type same as in cuda?

Reply:
Ascend NPU provides aclrtMallocHost API to allocate pin memory on host, which may improve performance in most scenarios. So that there is a case that data need be transferred between CANN EP and CPU EP during model running. So it's required.

Reply:
thanks for the explanation!